pkg/git/poll/handler.go

```diff
@@ -77,9 +78,13 @@ func (h *Handler) AddOrModifyGitRepoPollJob(ctx context.Context, gitRepo v1alpha
 			return
 		}
 		previousInterval := gitRepoPollJob.GitRepo.Spec.PollingInterval
+		previousGeneration := gitRepoPollJob.GitRepo.Generation
 		gitRepoPollJob.GitRepo = gitRepo
-		if gitRepo.Spec.PollingInterval != previousInterval {
-			// ignoring the error because we're going to schedule immediately
+		if (previousGeneration != gitRepo.Generation) ||
+			(!reflect.DeepEqual(gitRepo.Spec.PollingInterval, previousInterval)) {
```

Comment:
Why do we need to switch to using `reflect.DeepEqual` here? Did the previous, simpler comparison not work as expected?

Reply:
When working on https://github.com/rancher/fleet/pull/2537 I started to get weird errors when applying repos.
It was continuously creating the job, and it was because of that comparison.

Reply:
Because that field is a pointer:
```
type GitRepoSpec struct {
    ...
	// PollingInterval is how often to check git for new updates.
	PollingInterval *metav1.Duration `json:"pollingInterval,omitempty"`
```
I share @weyfonk's concerns here, I prefer avoiding reflection when possible. Unfortunately, unlike `metav1.Time`, `metav1.Duration` does not provide an `Equal` method, but we could use it as inspiration to implement our own:
https://github.com/kubernetes/apimachinery/blob/v0.30.2/pkg/apis/meta/v1/time.go#L74-L83

Reply:
I've added an `Equal` function to compare pointers to `v1.Duration` in the `durations` package.
I thought it makes sense to have that accesible to other packages